api/app/controllers/spree/api/taxons_controller.rb

```diff
@@ -106,6 +107,16 @@ def taxon_params
           {}
         end
       end
+
+      def preload_taxon_parents(taxons)
+        parents = Spree::Taxon.none
+
+        taxons.map do |taxon|
+          parents = parents.or(Spree::Taxon.ancestors_of(taxon))
+        end
+
+        Spree::Taxon.associate_parents(taxons + parents)
```

Comment:
Unfortunately, `self_and_ancestors` generate a new SQL query each time it's called and works on a taxon instance
In our case, it will do a query for each taxon we're loading and I'm trying to avoid exactly that 😕

Reply:
I might have spoken too fast. I'll try to use it instead of my new scope. I'll let you know what I find

Reply:
@kennyadsl : I confirm that we can reuse the method from awesome_nested_set. good catch !